engine/table/src/main/java/io/deephaven/engine/table/impl/TableDefaults.java

```diff
@@ -286,6 +292,23 @@ default Table formatColumns(String... columnFormats) {
         return updateView(selectColumns);
     }
 
+    @Override
+    @ConcurrentMethod
+    default Table formatDatabar(String columnName, String valueColumn, String axis, Double min, Double max,
+            String positiveColor, String negativeColor, String valuePlacement, String direction, Double opacity) {
+        String prefix = valueColumn + DATA_BAR_SUFFIX;
+        return this.naturalJoin(
+                this.aggBy(List.of(AggMin(prefix + "_MIN=" + valueColumn),
+                        AggMax(prefix + "_MAX=" + valueColumn)))
```

Comment:
Check if `min`/`max` are null first: if they're null, then use these aggregations (implicitly defining min/max based on the column); if they are not null, then just use the values they specified.